jme3-core/src/main/java/com/jme3/effect/shapes/EmitterSphereShape.java

```diff
@@ -95,11 +95,10 @@ public void cloneFields( Cloner cloner, Object original ) {
 
     @Override
     public void getRandomPoint(Vector3f store) {
-        do {
-            store.x = (FastMath.nextRandomFloat() * 2f - 1f) * radius;
-            store.y = (FastMath.nextRandomFloat() * 2f - 1f) * radius;
-            store.z = (FastMath.nextRandomFloat() * 2f - 1f) * radius;
-        } while (store.distance(center) > radius);
+        store.x = (FastMath.nextRandomFloat() * 2f - 1f) * radius;
+        store.y = (FastMath.nextRandomFloat() * 2f - 1f) * radius;
+        store.z = (FastMath.nextRandomFloat() * 2f - 1f) * radius;
+        store.addLocal(center);
```

Comment:
Now it creates a random point within a box of the desired side length (=radius) around the center.
The Do-while loop is still necessary to skip the samples outside the sphere

Reply:
mhhh... wouldn't it be simpler to do;
```
     store.x = FastMath.nextRandomFloat();
     store.y = FastMath.nextRandomFloat();
     store.z = FastMath.nextRandomFloat();
     store.normalizeLocal(); // <- random direction
     store.multLocal(radius); // <- mult by the radius to be inside the sphere radius
     store.addLocal(center); // <- relocal aroudn the center.
```
No loop needed and the normalize ensure that the point is inside the sphere.

Reply:
It would be simpler, but would not be uniformly distributed.

Reply:
Uh yeah first the x2-1 should stay to allow negative values. But then just take à random length between 0 and radius, and you break the uniformity.

Reply:
I might be wrong, but I think there is needs to more probability mass on the points with bigger radius than on the inner points. So if you normalize and then scale with a value between 0 and radius, that distribution must not be uniform, but something like u'=sqrt(1-u^2), as TehLeo suggested.